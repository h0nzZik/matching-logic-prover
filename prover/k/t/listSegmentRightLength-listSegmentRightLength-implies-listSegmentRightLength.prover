claim \implies( \and( listSegmentRightLength( variable("H") { ArrayIntInt }
                                            , variable("X") { Int } 
                                            , variable("Y") { Int } 
                                            , variable("FA") { Set } 
                                            , variable("LA") { Int }
                                            , .Patterns
                                            ) 
                    , listSegmentRightLength( variable("H") { ArrayIntInt } 
                                            , variable("Y") { Int } 
                                            , variable("Z") { Int } 
                                            , variable("FB") { Set } 
                                            , variable("LB") { Int }
                                            , .Patterns
                                            )
                    , \equals(variable("F") { Set }, union(variable("FA") { Set }, variable("FB") { Set }))
                    , disjoint(variable("FA") { Set }, variable("FB") { Set })
                    , \equals(variable("LENGTH") { Int }, plus(variable("LA") { Int }, variable("LB") { Int }))
                    , .Patterns
                    )
              , \and( listSegmentRightLength( variable("H") { ArrayIntInt } 
                                            , variable("X") { Int } 
                                            , variable("Z") { Int } 
                                            , variable("F") { Set } 
                                            , variable("LENGTH") { Int }
                                            , .Patterns
                                            )
                   , .Patterns
                   )
              ) 

strategy search-bound(5)

/*
strategy kt # index(1) # useAffectedHeuristic
         ; ( direct-proof
           | ( simplify ; direct-proof )
           | ( right-unfold-Nth(0, 1) ; simplify ; direct-proof ) 
           )
*/
