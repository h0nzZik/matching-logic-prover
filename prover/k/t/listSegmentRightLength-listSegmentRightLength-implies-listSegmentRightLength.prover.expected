<prover>
  <goal>
    <id>
      0
    </id>
    <active>
      true
    </active>
    <parent>
      -1
    </parent>
    <k>
      \implies ( \and ( listSegmentRightLength ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "Y" ) { Int } , variable ( "FA" ) { Set } , variable ( "LA" ) { Int } , .Patterns ) , listSegmentRightLength ( variable ( "H" ) { ArrayIntInt } , variable ( "Y" ) { Int } , variable ( "Z" ) { Int } , variable ( "FB" ) { Set } , variable ( "LB" ) { Int } , .Patterns ) , \equals ( variable ( "F" ) { Set } , union ( variable ( "FA" ) { Set } , variable ( "FB" ) { Set } ) ) , disjoint ( variable ( "FA" ) { Set } , variable ( "FB" ) { Set } ) , \equals ( variable ( "LENGTH" ) { Int } , plus ( variable ( "LA" ) { Int } , variable ( "LB" ) { Int } ) ) , .Patterns ) , \and ( listSegmentRightLength ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "Z" ) { Int } , variable ( "F" ) { Set } , variable ( "LENGTH" ) { Int } , .Patterns ) , .Patterns ) )
    </k>
    <strategy>
      success
    </strategy>
    <trace>
      .
    </trace>
  </goal>
</prover>
